gcore/src/prog.rs

```diff
@@ -18,84 +18,87 @@
 
 //! Program creation API for Gear programs.
 
-use crate::{error::Result, ActorId, CodeHash};
+use gear_core_errors::ExtError;
+
+use crate::{error::Result, ActorId, CodeId, MessageId};
 
 mod sys {
     use crate::error::SyscallError;
 
     extern "C" {
+        #[allow(improper_ctypes)]
         pub fn gr_create_program(
-            code_hash: *const u8,
+            code_id_ptr: *const [u8; 32],
             salt_ptr: *const u8,
             salt_len: u32,
-            data_ptr: *const u8,
-            data_len: u32,
-            value_ptr: *const u8,
-            program_id_ptr: *mut u8,
-            delay: *const u8,
+            payload_ptr: *const u8,
+            payload_len: u32,
+            value_ptr: *const u128,
+            delay: u32,
+            message_id_ptr: *mut [u8; 32],
+            program_id_ptr: *mut [u8; 32],
         ) -> SyscallError;
 
+        #[allow(improper_ctypes)]
         pub fn gr_create_program_wgas(
-            code_hash: *const u8,
+            code_id_ptr: *const [u8; 32],
```

Comment:
"code_hash" is common term, why introduce "id" here ?

"ptr" suffix is ok
```suggestion
            code_hash_ptr: *const [u8; 32],
```

Reply:
In gear-core we're going with hashes defined inside `MessageId`, `ProgramId` and `CodeId`, so I suggesting renaming of all `CodeHash` entries in order to support similar naming all over the project

Reply:
@grishasobol agreed with me.
Do you accept this solution, @NikVolf?

Reply:
Decided to leave it as it is atm.
To be changed with no logic changes at any time.